kernel/pci.c

```diff
@@ -12,17 +12,28 @@ typedef struct pci_handler_s {
   const char *name;
 } pci_handler;
 
+/* List of known class/subclasses, and how to handle them.  At some point this
+ * information may not be specific enough to determine how to use a given
+ * device, but it's okay for now. */
```

Comment:
I was figuring that there would a (sub)class handler that would then break out into a vendor/product specific handler if it was required for the individual device.

Perhaps a more right way to do this would be to add another two fields here for vendor and product, where if the entry is equal or negative one it is used, with the first match winning. This way vendor/product specific handlers could be added in addition to general class handlers.

As an aside, it might make sense to have some sort of PCI device struct that gets passed to the handler instead of a bunch of args, because the handler probably also wants the vendor/product information and probably other details from the device registers.